src/H4LTools.cc

```diff
@@ -0,0 +1,581 @@
+    float cutVal,mvaVal;
+    cutVal = 1000;
+    mvaVal = -1;
+    unsigned nE = (*nElectron).Get()[0];
+    for (unsigned int i=0; i<nE; i++){
+        if((*Electron_pt)[i]<10){
+            if(fabs((*Electron_eta)[i])<0.8) cutVal = 0.9044286167;
+            if((fabs((*Electron_eta)[i])>=0.8)&&(fabs((*Electron_eta)[i])<1.479)) cutVal = 0.9094166886;
+            if(fabs((*Electron_eta)[i])>=1.479) cutVal = 0.9443653660;
+        }
+        else{
+            if(fabs((*Electron_eta)[i])<0.8) cutVal = 0.1968600840;
+            if((fabs((*Electron_eta)[i])>=0.8)&&(fabs((*Electron_eta)[i])<1.479)) cutVal = 0.0759172100;
+            if(fabs((*Electron_eta)[i])>=1.479) cutVal = -0.5169136775;
+        }
+
+        mvaVal = (*Electron_mvaFall17V2Iso_WP90)[i];
+        if(mvaVal > cutVal){
+            tightid.push_back(true);
+        }
+        else{
+            tightid.push_back(false);
+        }
+    
+    }
+    
+    return tightid;
+    
+}
+std::vector<bool> H4LTools::passTight_Id(){
+    std::vector<bool> tightid;
+    unsigned nMu = (*nMuon).Get()[0];
+    for (unsigned int i=0; i<nMu; i++){
+        if (Muon_Pt_Corrected[i]<200){
+            tightid.push_back((*Muon_isPFcand)[i]);
+        }
+        else{
+            tightid.push_back((*Muon_isPFcand)[i]||((((*Muon_ptErr)[i]/Muon_Pt_Corrected[i])<0.3)&&(fabs((*Muon_dxy)[i])<0.2)&&(fabs((*Muon_dz)[i])<0.5)&&((*Muon_nTrackerLayers)[i]>5)));
+        }
+
+    }
+
+    return tightid;
+}
+
+std::vector<unsigned int> H4LTools::goodFsrPhotons(){
+    std::vector<unsigned int> goodFsrPhoton;
+    unsigned nFsr = (*nFsrPhoton).Get()[0];
+    for (unsigned int i=0; i<nFsr; i++){
+        if(((*FsrPhoton_pt)[i]>2)&&(fabs((*FsrPhoton_eta)[i])<2.4)&&((*FsrPhoton_relIso03)[i]<0.8)){
+            goodFsrPhoton.push_back(i);
+        }
+    }
+    return goodFsrPhoton;
+}
+
+std::vector<unsigned int> H4LTools::SelectedJets(std::vector<unsigned int> ele, std::vector<unsigned int> mu){
+    std::vector<unsigned int> goodJets;
+    unsigned nJ = (*nJet).Get()[0];
+    for(unsigned int i=0;i<nJ;i++){
+        int overlaptag=0;
+        TLorentzVector jettest;
+        jettest.SetPtEtaPhiM((*Jet_pt)[i],(*Jet_eta)[i],(*Jet_phi)[i],(*Jet_mass)[i]);
+        for(unsigned int ie=0;ie<ele.size();ie++){
+            TLorentzVector eletest;
+            eletest.SetPtEtaPhiM((*Electron_pt)[ele[ie]],(*Electron_eta)[ele[ie]],(*Electron_phi)[ele[ie]],(*Electron_mass)[ele[ie]]);
+            if(eletest.DeltaR(jettest)<0.4) overlaptag++;
+        }
+        for(unsigned int im=0;im<mu.size();im++){
+            TLorentzVector mutest;
+            mutest.SetPtEtaPhiM((*Muon_pt)[mu[im]],(*Muon_eta)[mu[im]],(*Muon_phi)[mu[im]],(*Muon_mass)[mu[im]]);
+            if(mutest.DeltaR(jettest)<0.4) overlaptag++;
+        }
+    
+      if(overlaptag==0){
+        if(((*Jet_pt)[i]>30)&&(fabs((*Jet_eta)[i])<4.7)){
+            std::cout<<"jetindex: "<<i<<"jetID "<<(*Jet_jetId)[i]<<" puID "<<(*Jet_puId)[i]<<std::endl;
```

Comment:
Comment this line or add the verbosity level so that we can switch on when we need to print this line.